pkgs/by-name/wl/wl-clicker/package.nix

```
@@ -0,0 +1,49 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  wayland-scanner,
+  wlr-protocols,
+  wayland,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wl-clicker";
+  version = "0.3.1";
+
+  nativeBuildInputs = [ wayland-scanner ];
+  buildInputs = [
+    wlr-protocols
+    wayland
+  ];
+
+  src = fetchFromGitHub {
+    owner = "phonetic112";
+    repo = "wl-clicker";
+    rev = "v${finalAttrs.version}";
+    sha256 = "0ddi383ilxqm0nvn05asn6m77sgi96fw07np95wx089a97b0mr8y";
+  };
+
+  postPatch = ''
+    sed -i 's|/usr|${wlr-protocols}|g' Makefile
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install build/wl-clicker $out/bin/wl-clicker
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Wayland autoclicker";
+    longDescription = "Script for auto clicking at incredibly high speeds - user must be a part of `input` group to run.";
+    homepage = "https://github.com/phonetic112/wl-clicker";
+    license = licenses.mit;
+    maintainers = [ maintainers.Flameopathic ];
+    mainProgram = "wl-clicker";
+    platforms = platforms.all;
```

It currently fails on aarch64-linux. We should either fix fix the build or mark as broken. Also, I assume 

```suggestion
    platforms = platforms.unix;
```

is more appropriate here.

---

would `platforms = lib.intersectLists (lib.platforms.linux) (lib.platforms.x86_64);` work for solving these issues, for now? I don't think i'm at the stage where i know what's going wrong so i can fix it.

---

If the package is supposed to work on said platforms, but is broken, we should use `broken = ...;`. If it is not supposed to work at all, then set it in platforms.

If it doesn't build on darwin (I haven't checked) and aarch64-linux, you can use `broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;`

---

okay, thank you.